pytorch_lightning/trainer/properties.py

```diff
@@ -480,6 +483,13 @@ def sanity_checking(self, val: bool) -> None:
     Loop properties
     """
 
+    @property
+    def active_loop(self) -> Optional[Union[TrainLoop, EvaluationLoop]]:
+        if self.training:
+            return self.train_loop
+        elif self.sanity_checking or self.evaluating:
+            return self.evaluation_loop
+
```

Comment:
why does this need to be exposed as a property? doesn't this leak the implementation detail? what if someone accesses the active_loop and then modifies properties on it?

Reply:
So we want to access the current `ResultCollection` object from the logger connector.
And each `Loop` has its own `ResultCollection`.
So we need this to get the current running loop.

I guess we can make this property protected to discourage external modifications.

cc: @awaelchli

Reply:
do you want me to directly do it now in the new loops? should be no problem.

Reply:
Did it already with https://github.com/PyTorchLightning/pytorch-lightning/pull/7891/commits/ab28850d71363c960c4ad5d20431b9ca01d9a3e1

Reply:
so this was not about the results in loops?